neuro_data/movies/stats.py

```diff
@@ -217,4 +218,212 @@ def make(self, key):
         key['min_trial_repeats'] = counts[mask].min()
         key['min_frames'] = np.min([dataset[index].responses.shape[0] for index in np.where(np.isin(dataset_condition_hashes, condition_hashes))[0]])
 
-        self.insert1(key)
\ No newline at end of file
+        self.insert1(key)
+
+@schema
+class BootstrapOracleSeed(dj.Lookup):
+    definition = """
+    oracle_bootstrap_seed                 :  int # random seed
+    ---
+    """
+
+    @property
+    def contents(self):
+        for seed in list(range(100)):
+            yield (seed,)
+
+@schema
+class BootstrapOracle(dj.Computed):
+    definition = """
+    -> OracleStims
+    -> BootstrapOracleSeed
+    ---
+    """
+
+    class Score(dj.Part):
+        definition = """
+        -> master
+        ---
+        boostrap_score_true			    : float
+        boostrap_score_null			    : float
+        """
+
+    class UnitScore(dj.Part):
+        definition = """
+        -> master
+        -> MovieScan.Unit
+        ---
+        boostrap_unit_score_true		: float
+        boostrap_unit_score_null		: float
+        """
+
+    def sample_from_condition_hash(self, target_hash, dataset, sample_size):
+        return np.random.choice(np.where(dataset == target_hash)[0], sample_size, replace=False)
+
+    def check_input(self, target_indices, dataset):
+        dataset_images_shape = dataset[target_indices[0]].inputs.shape
+        inputs = np.empty(shape=[len(target_indices), dataset_images_shape[0], dataset_images_shape[1], dataset_images_shape[2], dataset_images_shape[3]])
+        for i, index in enumerate(target_indices):
+            inputs[i] = dataset[index].inputs
+        assert np.all(np.abs(np.diff(inputs, axis=0)) == 0), 'Images of oracle trials do not match'
+
+    def sample_frames_from_dataset(self, index, dataset, min_frames, num_of_neurons):
+        # Compute start_index
+        starting_index = 0
+        upper_bound = dataset[index].responses.shape[0] - min_frames
+    
+        if upper_bound != 0:
+            starting_index = np.random.randint(0, upper_bound)
+            
+        # Sampling min_frames from dataset
```

Comment:
Don't use this for-loop for indexing.  Instead do:

frame_responses = dataset[index].responses[starting_index:starting_index+min_frames]